SystemWrapper.Tests/DateTimeWrapTests.cs

```diff
@@ -0,0 +1,61 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SystemInterface;
+
+namespace SystemWrapper.Tests
+{
+    [TestClass]
+    public class DateTimeWrapTests
+    {
+        const long dateTicks = 636846612300000000L;
+
+        [TestMethod]
+        public void Equatable_SameDateTime()
+        {
+            IDateTime date1 = new DateTimeWrap(dateTicks);
+            IDateTime date2 = new DateTimeWrap(dateTicks);
+            object dateObject = new DateTimeWrap(dateTicks);
+
+            Assert.AreEqual(date1, date1);
+            Assert.AreEqual(date1, date2);
+            Assert.AreEqual(date1, dateObject);
+            Assert.AreEqual(date1.GetHashCode(), date2.GetHashCode());
```

Comment:
I've assumed that we're only interested in comparing an `IDateTime` with another `IDateTime`. I've not considered the case where we may want to compare an `IDateTime` with a .NET `DateTime` instance. If it's important to support this use case, then let me know and I'll add support for it.